compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1715,13 +1715,11 @@ impl<'a> Parser<'a> {
                     // the AST for typechecking.
                     err.span_label(ident.span, "while parsing this `fn`");
                     err.emit();
-                    (Vec::new(), None)
                 } else {
                     return Err(err);
                 }
-            } else {
-                unreachable!()
```

Comment:
maybe add a `delay_span_bug` to protect against this ever passing compilation.

Reply:
esteban and I had a short chat. This is ok, even if not idea. There should be an issue mentioning the wrong first error suggestion (as seen in the newly added test) and that we should find a way to avoid that message and just show the second one.

r=me with that issue opened.